storage/simulation/cluster.go

```diff
@@ -104,29 +113,102 @@ func (c *Cluster) addStore(nodeID proto.NodeID) *Store {
 // store.
 func (c *Cluster) addRange() *Range {
 	rangeID := proto.RangeID(len(c.ranges))
-	newRng := newRange(rangeID)
+	newRng := newRange(rangeID, c.allocator)
 	c.ranges[rangeID] = newRng
 	return newRng
 }
 
+// splitRangeRandom splits a random range from within the cluster.
 func (c *Cluster) splitRangeRandom() {
 	rangeID := proto.RangeID(c.rand.Int63n(int64(len(c.ranges))))
 	c.splitRange(rangeID)
 }
 
+// splitRangeLast splits the last added range in the cluster.
 func (c *Cluster) splitRangeLast() {
 	rangeID := proto.RangeID(len(c.ranges) - 1)
 	c.splitRange(rangeID)
 }
 
+// splitRange "splits" a range. This split creates a new range with new
+// replicas on the same stores as the passed in range. The new range has the
+// same zone config as the original range.
 func (c *Cluster) splitRange(rangeID proto.RangeID) {
 	newRange := c.addRange()
 	originalRange := c.ranges[rangeID]
 	newRange.splitRange(originalRange)
 }
 
+// runEpoch steps through a single instance of the simulator. Each epoch
+// performs the following steps:
+// 1) The status of every store is gossiped so the store pool is up to date.
+// 2) Each replica on every range calls the allocator to determine if there are
+//    any actions required.
+// 3) The replica on each range with the highest priority executes it's action.
+// 4) The current status of the cluster is output.
+func (c *Cluster) runEpoch() {
+	c.epoch++
+
+	// Gossip all the store updates.
+	c.gossipStores()
+
+	// Determine next operations for all ranges.
```

Comment:
i'd expand on this comment a bit to explain why it's inappropriate to simple execute each action after determining it.